packages/contracts-bsc/contracts/BscTokenController.sol

```diff
@@ -0,0 +1,692 @@
+    */
+
+    /**
+     * @dev Throws if called by any account other than the owner.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, "only Owner");
+        _;
+    }
+
+    /**
+     * @dev Modifier throws if called by any account other than the pendingOwner.
+     */
+    modifier onlyPendingOwner() {
+        require(msg.sender == pendingOwner);
+        _;
+    }
+
+    /**
+     * @dev Modifier throws if called by any account other than proofOfReserveEnabler or owner.
+     */
+    modifier onlyProofOfReserveEnablerOrOwner() {
+        require(msg.sender == proofOfReserveEnabler || msg.sender == owner, "only proofOfReserveEnabler or owner");
+        _;
+    }
+
+    /**
+     * @dev sets the original `owner` of the contract to the sender
+     * at construction. Must then be reinitialized
+     */
+    constructor() public {
+        owner = msg.sender;
+        emit OwnershipTransferred(address(0), owner);
+    }
+
+    function initialize() public {
+        require(!initialized, "already initialized");
+        owner = msg.sender;
+        emit OwnershipTransferred(address(0), owner);
+
+        instantMintThreshold = 150_000_000_000_000_000_000_000_000; // 150 M
+        ratifiedMintThreshold = 300_000_000_000_000_000_000_000_000; // 300 M
+        multiSigMintThreshold = 1_000_000_000_000_000_000_000_000_000; // 1 B
+        emit MintThresholdChanged(
+            150_000_000_000_000_000_000_000_000,
+            300_000_000_000_000_000_000_000_000,
+            1_000_000_000_000_000_000_000_000_000
+        );
+        instantMintLimit = 150_000_000_000_000_000_000_000_000; // 150 M
+        ratifiedMintLimit = 300_000_000_000_000_000_000_000_000; // 300 M
+        multiSigMintLimit = 1_000_000_000_000_000_000_000_000_000; // 1 B
+        emit MintLimitsChanged(
+            150_000_000_000_000_000_000_000_000,
+            300_000_000_000_000_000_000_000_000,
+            1_000_000_000_000_000_000_000_000_000
+        );
+
+        initialized = true;
+    }
+
+    /**
+     * @dev Allows the current owner to set the pendingOwner address.
+     * @param newOwner The address to transfer ownership to.
+     */
+    function transferOwnership(address payable newOwner) external onlyOwner {
+        pendingOwner = newOwner;
+        emit NewOwnerPending(address(owner), address(pendingOwner));
+    }
+
+    /**
+     * @dev Allows the pendingOwner address to finalize the transfer.
+     */
+    function claimOwnership() external onlyPendingOwner {
+        emit OwnershipTransferred(address(owner), address(pendingOwner));
+        owner = pendingOwner;
+        pendingOwner = address(0);
+    }
+
+    /*
+    ========================================
+    proxy functions
+    ========================================
+    */
+
+    function transferTrueCurrencyProxyOwnership(address _newOwner) external onlyOwner {
+        OwnedUpgradeabilityProxy(address(uint160(address(token)))).transferProxyOwnership(_newOwner);
```

Comment:
When using `OUP(address(token))`, a TypeError will be raised with the message 'Explicit type conversion not allowed from non-payable "address" to "contract OwnedUpgradeabilityProxy", which has a payable fallback function.' Therefore, to avoid this error, you can either use `OwnedUpgradeabilityProxy(address(uint160(address(token))))` or use the interface `IOwnedUpgradeabilityProxy(address(token))`.